src/DataWrangling/EN4/EN4.jl

```diff
@@ -0,0 +1,95 @@
+module EN4
+
+export EN4Metadatum, EN4_immersed_grid, adjusted_EN4_tracers, initialize!
+export EN4Monthly
+export EN4FieldTimeSeries, EN4Restoring, LinearlyTaperedPolarMask
+
+using ClimaOcean
+using ClimaOcean.DataWrangling
+using ClimaOcean.DataWrangling: inpaint_mask!, NearestNeighborInpainting,
+                                download_progress, compute_native_date_range,
+                                shift_longitude_to_0_360, Kelvin, Celsius
+
+using Oceananigans
+using Oceananigans.Architectures: architecture, child_architecture
+using Oceananigans.BoundaryConditions
+using Oceananigans.DistributedComputations
+using Oceananigans.DistributedComputations: DistributedField, all_reduce, barrier!
+using Oceananigans.Utils
+
+using KernelAbstractions: @kernel, @index
+using NCDatasets
+using JLD2
+using Downloads: download
+using Dates
+using Adapt
+using Scratch
+
+import ClimaOcean.DataWrangling: vertical_interfaces, empty_field, variable_is_three_dimensional,
+                                 inpainted_metadata_path
+
+download_EN4_cache::String = ""
+function __init__()
+    global download_EN4_cache = @get_scratch!("EN4")
+end
+
+include("EN4_metadata.jl")
+include("EN4_mask.jl")
+
+vertical_interfaces(metadata::Metadata{<:EN4Monthly}) =
+    [
+    -5500.0,
+    -5200.5986,
+    -4901.459,
+    -4602.695,
+    -4304.469,
+    -4007.0154,
+    -3710.6636,
+    -3415.8828,
+    -3123.3313,
+    -2833.926,
+    -2548.9243,
+    -2270.0168,
+    -1999.4135,
+    -1739.8875,
+    -1494.7288,
+    -1267.542,
+    -1061.846,
+    -880.5102,
+    -725.1709,
+    -595.8501,
+    -490.9494,
+    -407.6244,
+    -342.3651,
+    -291.5638,
+    -251.9186,
+    -220.6418,
+    -195.5097,
+    -174.8189,
+    -157.3019,
+    -142.0345,
+    -128.3522,
+    -115.7825,
+    -103.9913,
+    -92.7437,
+    -81.8753,
+    -71.2711,
+    -60.8509,
+    -50.5586,
+    -40.3553,
+    -30.214,
+    -20.1158,
+    -10.0475,
+      0.0,
+    ]
+
+function inpainted_metadata_filename(metadata::EN4Metadata)
+    original_filename = metadata_filename(metadata)
+    without_extension = original_filename[1:end-3]
+    var = string(metadata.name)
+    return without_extension * "_" * var *"_inpainted.jld2"
```

Comment:
I think this function is general and should be moved up one level -- the only difference is that for EN4: we `return without_extension * "_" * var *"_inpainted.jld2"` and for ECCO: we `return without_extension * "_inpainted.jld2"`.

The inclusion of `var` is more general as it works for both cases where variables are in separate files (as in ECCO) and where variables are in the one file (as in EN4).

Reply:
I'll try it and see what happens with eg ECCO. Would we have filenames like `THETA_xyz_temperature_inpainted.jld2`?

Reply:
(shouldn't matter even if this is the case because in an ideal world all of these files are somewhere in a scratch directory and not visible to the user?)

Reply:
Yes that is what the ECCO filename would look like I think!